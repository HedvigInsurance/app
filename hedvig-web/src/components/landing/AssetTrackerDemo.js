import React from "react"
import {
  AssetTrackerDemoStyled,
  AssetTrackerDemoTextContainer,
  AssetTrackerDemoPhoneContainer,
  AssetTrackerDemoPhone,
  CenteredColumn
} from "../styles/landing"
import { BlackPurpleHeading2 } from "../styles/typography"
import { WhiteRoundedButton } from "../Button"

const MyBlackPurpleHeading2 = BlackPurpleHeading2.extend`
  text-align: center;
  @media (min-width: 1024px) {
    text-align: left;
  }
`

const MyCenteredColumn = CenteredColumn.extend`
  @media (min-width: 992px) {
    align-items: flex-start;
  }
`

export default class AssetTrackerDemo extends React.Component {
  state = {
    expanded: false
  }

  unexpanded() {
    return (
      <MyCenteredColumn>
        <WhiteRoundedButton onClick={() => this.setState({ expanded: true })}>
          Mer om prylbanken
        </WhiteRoundedButton>
      </MyCenteredColumn>
    )
  }

  expanded() {
    return (
      <div>
        <p style={{ marginTop: -20 }}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
          eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
          minim veniam, quis nostrud exercitation ullamco laboris nisi ut
          aliquip ex ea commodo consequat. Duis aute irure dolor in
          reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
          pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
          culpa qui officia deserunt mollit anim id est laborum.
        </p>
        <MyCenteredColumn>
          <WhiteRoundedButton
            onClick={() => this.setState({ expanded: false })}
          >
            Mindre om prylbanken
          </WhiteRoundedButton>
        </MyCenteredColumn>
      </div>
    )
  }

  maybeExpanded() {
    return this.state.expanded ? this.expanded() : this.unexpanded()
  }

  render() {
    return (
      <AssetTrackerDemoStyled className="pure-g">
        <AssetTrackerDemoPhoneContainer className="pure-u-lg-1-2 pure-u-sm-1-1 pure-u-1-1">
          <AssetTrackerDemoPhone />
        </AssetTrackerDemoPhoneContainer>
        <AssetTrackerDemoTextContainer className="pure-u-lg-1-2 pure-u-sm-1-1 pure-u-1-1">
          <MyBlackPurpleHeading2>
            Låt Hedvig hålla koll på dina prylar, och se exakt hur de är
            försäkrade.
          </MyBlackPurpleHeading2>
          {this.maybeExpanded()}
        </AssetTrackerDemoTextContainer>
      </AssetTrackerDemoStyled>
    )
  }
}
