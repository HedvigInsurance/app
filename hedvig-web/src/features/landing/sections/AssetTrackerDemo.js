import React from "react"
import styled from "styled-components"
import {
  CenteredColumn,
  HeadingSubText,
  AbsoluteFadeInParagraph
} from "../../../components/styles/landing"
import { BlackPurpleHeading2 } from "../../../components/styles/typography"
import { WhiteRoundedButton } from "../../../components/Button"

const MyBlackPurpleHeading2 = BlackPurpleHeading2.extend`
  text-align: center;
  padding: 1em 1em 0;
  font-size: 28px;
  line-height: 38px;
  @media (min-width: 800px) {
    text-align: left;
    padding: 1em 0 0.3em;
    font-size: 40px;
    line-height: 56px;
  }
`

const MyCenteredColumn = CenteredColumn.extend`
  display: none;

  @media (min-width: 800px) {
    display: flex;
    align-items: flex-start;
  }
`

const Container = styled.div`
  display: flex;
  background-color: white;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  min-height: 100%;
  margin: 3em 0;

  @media (min-width: 800px) {
    flex-direction: row-reverse;
    justify-content: space-around;
  }
`

const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 800px;
  padding: 0 0 1em;

  @media (min-width: 800px) {
    align-items: flex-start;
    padding: 0;
  }
`

const PhoneContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`

const Phone = styled.img`
  width: 320px;
  height: 379px;

  @media (min-width: 800px) {
    width: 440px;
    height: 521px;
  }
`

const PreferredBreak = styled.br`
  display: none;

  @media (max-width: 320px) {
    display: inline;
  }
`

const PreferredBreakOnLarge = styled.br`
  display: none;
  @media (min-width: 800px) {
    display: inline;
  }
`

export default class AssetTrackerDemo extends React.Component {
  state = {
    expanded: false
  }

  maybeExpanded() {
    return (
      <div>
        <MyCenteredColumn>
          <WhiteRoundedButton
            onClick={() => this.setState({ expanded: !this.state.expanded })}
            style={{minWidth: "240px", color: this.state.expanded ? "white" : "#651EFF" , backgroundColor: this.state.expanded ? "#651EFF" : "white" }}
          >
            { this.state.expanded ? "Mindre om prylbanken" : "Mer om prylbanken" }
          </WhiteRoundedButton>
        </MyCenteredColumn>
        { this.state.expanded ? (
          <AbsoluteFadeInParagraph>
            Med prylbanken blir det enkelt att hålla ordning på sakerna du bryr dig om.
            Kvitton och viktiga papper är säkrare hos Hedvig än i en låda under sängen,
            och du ser hur försäkringen gäller för varenda pryl.
            Om olyckan är framme anmäler du skadan med ett knapptryck.
          </AbsoluteFadeInParagraph>
        ) : null }
      </div>
    )
  }

  render() {
    return (
      <Container>
        <TextContainer>
          <MyBlackPurpleHeading2>Låt Hedvig hålla koll på dina saker,<PreferredBreakOnLarge /> se exakt hur de är försäkrade</MyBlackPurpleHeading2>
          <HeadingSubText>
            Drulle ingår för alla<PreferredBreak/> prylar värda under 50 000 kr
          </HeadingSubText>
          {this.maybeExpanded()}
        </TextContainer>
        <PhoneContainer>
          <Phone
            src="/assets/web/Images/Hedvig_Prylbanken.png"
            alt="Hedvigs prylbank"
          />
        </PhoneContainer>
      </Container>
    )
  }
}
