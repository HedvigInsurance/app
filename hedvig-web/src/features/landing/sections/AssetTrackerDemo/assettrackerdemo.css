.AssetTrackerDemo {
  height: auto;
}


.AssetTrackerDemo__text {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100%;
}

.AssetTrackerDemo__heading {
  font-size: 28px;
  font-family: "Merriweather";
  font-weight: 400;
  line-height: normal;
}

.AssetTrackerDemo__phone {
  width: 320px;
  height: 379px;
}

@media screen and (min-width: 768px) {
  .pure-g.AssetTrackerDemo__column-order {
    flex-direction: row-reverse;
  }

  .AssetTrackerDemo__text {
    text-align: start;
  }

  .AssetTrackerDemo__phone {
    width: 440px;
    height: 521px;
  }
}
