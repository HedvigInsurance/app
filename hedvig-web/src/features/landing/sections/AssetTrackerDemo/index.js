import React from "react"
import { WhiteRoundedButton } from "../../../../components/Button"

import "./assettrackerdemo.css"

export default class AssetTrackerDemo extends React.Component {
  state = {
    expanded: false
  }

  maybeExpanded() {
    return (
      <div>
        <div className="AssetTrackerDemo__expandable">
          <WhiteRoundedButton
            onClick={() => this.setState({ expanded: !this.state.expanded })}
            style={{minWidth: "240px", color: this.state.expanded ? "white" : "#651EFF" , backgroundColor: this.state.expanded ? "#651EFF" : "white" }}
          >
            { this.state.expanded ? "Mindre om prylbanken" : "Mer om prylbanken" }
          </WhiteRoundedButton>
        </div>
        { this.state.expanded ? (
          <p className="AssetTrackerDemo__fade-in-paragraph">
            Med prylbanken blir det enkelt att hålla ordning på sakerna du bryr dig om.
            Kvitton och viktiga papper är säkrare hos Hedvig än i en låda under sängen,
            och du ser hur försäkringen gäller för varenda pryl.
            Om olyckan är framme anmäler du skadan med ett knapptryck.
          </p>
        ) : null }
      </div>
    )
  }

  render() {
    return (
      <section className="pure-g pure-centered AssetTrackerDemo">
        <div className="pure-u-1-1 pure-u-lg-22-24">
          <div className="pure-g AssetTrackerDemo__column-order">
            <div className="pure-u-1-1 pure-u-lg-3-5">
              <div className="AssetTrackerDemo__text">
                <h1 className="AssetTrackerDemo__heading">Låt Hedvig hålla koll på dina saker, se exakt hur de är&nbsp;försäkrade</h1>
                <p>
                  Drulle ingår för alla prylar värda under 50 000 kr
                </p>
                {this.maybeExpanded()}
              </div>
            </div>
            <div className="pure-u-1-1 pure-u-lg-2-5 AssetTrackerDemo__phone-container">
              <img
                className="AssetTrackerDemo__phone"
                src="/assets/web/Images/Hedvig_Prylbanken.png"
                alt="Hedvigs prylbank"
              />
            </div>
          </div>
        </div>
      </section>
    )
  }
}
