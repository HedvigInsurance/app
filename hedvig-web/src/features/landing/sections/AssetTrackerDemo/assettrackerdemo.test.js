import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import AssetTrackerDemo from "./index"

describe('<AssetTrackerDemo />', () => {
  it('Should render correctly', () => {
    const component = shallow(<AssetTrackerDemo />)

    expect(toJson(component)).toMatchSnapshot()
  })
})
