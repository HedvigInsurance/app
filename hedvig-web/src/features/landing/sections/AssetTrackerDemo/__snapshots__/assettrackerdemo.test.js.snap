// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssetTrackerDemo /> Should render correctly 1`] = `
<section
  className="pure-g pure-centered AssetTrackerDemo"
>
  <div
    className="pure-u-1-1 pure-u-lg-22-24"
  >
    <div
      className="pure-g AssetTrackerDemo__column-order"
    >
      <div
        className="pure-u-1-1 pure-u-lg-3-5"
      >
        <div
          className="AssetTrackerDemo__text"
        >
          <h1
            className="AssetTrackerDemo__heading"
          >
            Låt Hedvig hålla koll på dina saker, se exakt hur de är försäkrade
          </h1>
          <p>
            Drulle ingår för alla prylar värda under 50 000 kr
          </p>
          <div>
            <div
              className="AssetTrackerDemo__expandable"
            >
              <WhiteRoundedButton
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "white",
                    "color": "#651EFF",
                    "minWidth": "240px",
                  }
                }
              >
                Mer om prylbanken
              </WhiteRoundedButton>
            </div>
          </div>
        </div>
      </div>
      <div
        className="pure-u-1-1 pure-u-lg-2-5 AssetTrackerDemo__phone-container"
      >
        <img
          alt="Hedvigs prylbank"
          className="AssetTrackerDemo__phone"
          src="/assets/web/Images/Hedvig_Prylbanken.png"
        />
      </div>
    </div>
  </div>
</section>
`;
